localstack/services/awslambda/lambda_api.py

```diff
@@ -1153,6 +1158,9 @@ def create_function():
         result.update(format_func_details(lambda_function))
         if data.get("Publish"):
             result["Version"] = publish_new_function_version(arn)["Version"]
+        LAMBDA_ASYNC_LOCKS.assure_lock_present(
+            lambda_name, threading.BoundedSemaphore(100)
```

Comment:
100 should be a "safe" default for now I guess but we can also explicitly handle it being None instead.